src/compiler/crystal/compiler.cr

```diff
@@ -441,9 +441,7 @@ module Crystal
       puts
       puts "Macro runs:"
       program.compiled_macros_cache.each do |filename, compiled_macro_run|
-        print " - "
-        print filename
-        print ": "
+        print " - #{filename}: "
```

Comment:
Not very important in this case (not a potential tight loop) but the original 3 prints pushed 2 static strings and 1 dynamically allocated string to the output buffer, whereas your change now allocates another dynamic string and eventually pushes it to the output buffer, hence requires a thrown-away allocation.

Reply:
I also made a benchmark
```Crystal
require "benchmark"

filename = "file.cr"

Benchmark.ips do |x|
  x.report "3 prints" do
    print " - "
    print filename
    print ": "
  end

  x.report "1 print" do
    print " - #{filename}: "
  end
end
```
and the 1 print is 2 times faster.
```
3 prints   8.35k (119.77µs) (± 6.16%)  2.30× slower
 1 print  19.17k ( 52.16µs) (± 2.83%)       fastest
```

Reply:
What the.... Why is that faster? @ysbaddaden you know?

Reply:
It's related to IO being flushed, if you use another function call IO stops being the bottleneck:

```crystal
require "benchmark"

def print2(param); end

filename = "file.cr"

Benchmark.ips do |x|
  x.report "3 prints" do
    print2 " - "
    print2 filename
    print2 ": "
  end

  x.report "1 print" do
    print2 " - #{filename}: "
  end
end
```

```
3 prints 442.72M (  2.26ns) (± 4.78%)       fastest
 1 print   6.52M (153.48ns) (± 4.70%) 67.95× slower
```

Reply:
Fastest is probably `print " - ", filename, ": "`

Reply:
You are right. It's actually faster.
```Crystal
require "benchmark"

filename = "file.cr"

Benchmark.ips do |x|
  x.report "3 prints" do
    3.times do
      print " - "
      print filename
      print ": "
    end
  end

  x.report "interpolation print" do
    3.times do
      print " - #{filename}: "
    end
  end

  x.report "comma print" do
    3.times do
      print " - ", filename, ": "
    end
  end
end
```
(I used `3.times` to have clearer results)
```
           3 prints   2.47k (404.48µs) (±10.79%)  2.53× slower
interpolation print   5.75k (173.88µs) (±11.99%)  1.09× slower
        comma print   6.25k (160.11µs) (±13.01%)       fastest
```

But i'm sure there are many more interpolation prints like this. Does that means every interpolation print should be changed to a comma print instead?

Reply:
Please, use `print " - ", filename, ": "` version.